### DISCUSSION TITLE
Build appimage from rootfs archive distro.

### DISCUSSION THREAD

**Discussion by lidgnulinux** [AppImage/AppImageKit#1361] [Q&A] [answered]
So I'm trying to convert rootfs archive (alpine linux rootfs) into appimage. I followed this [guide](https://github.com/AppImage/AppImageKit/wiki/Bundling-command-line-tools) and succeeded. Appimage was created. When I ran the appimage : 

```
$ test_x86_64.AppImage command
```

There are some missing libraries which already exist on rootfs archive. How to tell the appimage to use libraries from rootfs ?

Note : I want to create minimalistic neovim appimage using alpine rootfs archive and appimagetool. These are steps I did : 

1. Unpack older alpine linux minirootfs, I got it from [here](https://dl-cdn.alpinelinux.org/alpine/v3.17/releases/x86_64/).
1. install neovim in rootfs.

    ```
    $ sudo chroot /home/user/rootfs_dir/ /sbin/apk add neovim
    ```
1. Copy `AppRun` to rootfs_dir, I got the script from [here](https://github.com/AppImage/AppImageKit/wiki/Bundling-command-line-tools).
1. Add icon, desktop file to rootfs_dir
1. Generate appimage using appimage tool.

    ```
    $ appimagetool-x86_64.AppImage -v -n ./rootfs_dir
    ```
1. Test the generated appimage.

   ```
    $ ./generated_appimage nvim
    ```
I got error about missing libraries for neovim executable. Any clue to fix it ? Thanks.

**Comment by lidgnulinux** [comment]
Found related discussion [here](https://github.com/orgs/AppImage/discussions/1297). Tried to patchelf the nvim executable but missing libraries error still appear.

**Comment by probonopd** [comment]
I've experimented with doing a similar thing here:
https://github.com/probonopd/alpine2appimage

**Reply by lidgnulinux** [reply]
Hi @probonopd  . Finally, I made it !
Instead of AppRun script, I use AppRun executable. I stiil need to link /lib/ld-musl-x86_64.so.1 to /lib64/ld-linux-x86-64.so.2. Now my generated appimage works. Thanks.

**Comment by lidgnulinux** [comment, marked as answer]
Finally, I made it !
Instead of AppRun script, I use AppRun executable. I stiil need to link `/lib/ld-musl-x86_64.so.1` to `/lib64/ld-linux-x86-64.so.2`. Now my generated appimage works. Thanks.

**Comment by lidgnulinux** [comment]
Since it's solved, I will close it for now.
